### PR TITLE
New script update-from-local.sh that supports local development; similar to update-from-devvm.sh

### PR DESCRIPTION
**Pull request by sfc-gh-azwiegincew**

### CODE REVIEW

File: src/snowflake/snowpark/_internal/proto/update-from-local.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  then
+    echo "Using source root \"$1\""
+    SRC_ROOT=$1
+fi
+set -euxo pipefail
+
+SCRIPT_DIR=$(dirname "$0")
+
+# Step 1: Build the python proto file from scratch via bazel
+pushd $SRC_ROOT/Snowpark
```
Comment (sfc-gh-lspiegelberg) [inline src/snowflake/snowpark/_internal/proto/update-from-local.sh:21]:
```pushd ... > /dev/null && \
./make_ast.sh && popd
```
else, when make_ast.sh fails you're stuck in the wrong dir...

Comment (sfc-gh-azwiegincew) [reply src/snowflake/snowpark/_internal/proto/update-from-local.sh:21]: Isn't it the opposite? We want to popd in all cases, not just when make_ast succeeds.

File: src/snowflake/snowpark/_internal/proto/update-from-local.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  then
+    echo "No source directory provided, defaulting to \"$SRC_ROOT\""
+elif [ -n "$1" ]
+  then
+    echo "Using source root \"$1\""
+    SRC_ROOT=$1
+fi
+set -euxo pipefail
+
+SCRIPT_DIR=$(dirname "$0")
```
Comment (sfc-gh-lspiegelberg) [inline src/snowflake/snowpark/_internal/proto/update-from-local.sh:18]: To execute relative, this is pattern
```
# execute relative to script
SCRIPT_LOCATION=$(dirname "$(readlink -f "$BASH_SOURCE")")
cd $SCRIPT_LOCATION
```

**Review by sfc-gh-lspiegelberg — APPROVED** [review]
couple nits, mainly for some weird bash edge cases.

**Pull request merged**